File: Algorithms/Hashing, Hashtable or Hashmap, Pre-computation/Problem Solved/1_LC1_Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        elementsStore = {}

        for i in range(n):
            remain = target - nums[i]
            if remain in elementsStore:
                return [i, elementsStore[remain]]
            elementsStore[nums[i]] = i

File: Algorithms/Hashing, Hashtable or Hashmap, Pre-computation/Problem Solved/1_LC1_Two_Sum.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        int n = nums.size();
        map<int, int> elementStore;

        for (int i = 0; i < n; ++i)
        {
            int remain = target - nums[i];
            if (elementStore.find(remain) != elementStore.end())
                return {i, elementStore[remain]};
            else
                elementStore[nums[i]] = i;
        }

        return {-1, -1};
    }
};

int main()
{

    return 0;
}